Student Sketches and Answers.jl

```diff
@@ -27,238 +26,285 @@ begin
+            if(size(student_answers, 1) == 2 || student_answers.q_id[1] == q_id_2)
+                end_timestamp_2 = student_answers[student_answers.q_id .== q_id_2, :].timestamp[1]
+            else
+			end_timestamp_2 = sketch_data.timestamp[end]
+            end
+            animation_data = filter(row -> (end_timestamp_1 <= row.timestamp) && (row.timestamp <= end_timestamp_2), sketch_data)
+        end
+
+		#columns for x, y and timestamp
+        x = animation_data.xcor
+        y = animation_data.ycor
+        t = animation_data.timestamp
+
+        for i in 1:size(animation_data, 1) 
+			#updating the y vector using x and y data points
+            current_sketch_y_vector[x[i] + 1] = y[i] 
+			#pushing the uid, timestamp and the copy of updated y vector in the dataframe
+            push!(sy_data.uid, student)
+            push!(sy_data.timestamp, t[i])
+            push!(sy_data.y_vector, copy(current_sketch_y_vector))
+        end
+    end
+
+    for uid in unique(a_df.uid)
+		#ignoring the uids with no sketch data
+        if !any(s_df[:, :uid] .== uid)
+            continue
+        end
+		#filtering the datsets for the selected student
+        student_answers = a_df[a_df.uid .== uid, :]
+        sketch_data = s_df[s_df.uid .== uid, :]
+		#calling the helepr function to update the y vectors for the selected student
+        update_sy_vector(student_answers, q_id_1, sketch_data, sy_data_1, uid)
+        update_sy_vector(
+			student_answers, 
+			q_id_2, 
+			sketch_data, 
+			sy_data_2, 
+			uid,
+			deepcopy(sy_data_1[sy_data_1.uid  .== uid, :].y_vector[end]))
+    end
+
+    return sy_data_1, sy_data_2
+end
+
+
+# ╔═╡ 5a1f1db0-bc13-4e3b-ac5a-ba63b71b5ba3
+begin
+	#filtered answer data frame containing the entries of the selected student
+	student_answers = a_df[a_df.uid .== selected_student, :]    
+	if(size(student_answers, 1) == 2 || student_answers.q_id[1] == q_id_1)
+		sketch_url_q_id_1 = student_answers[student_answers.q_id .== q_id_1, :].sketch_url[1]; #url for Q1
+		text_answer_q_id_1 = student_answers[student_answers.q_id .== q_id_1, :].text_answer[1]; #ans for Q1
+	end
+	if(size(student_answers, 1) == 2 || student_answers.q_id[1] == q_id_2)
+		sketch_url_q_id_2 = student_answers[student_answers.q_id .== q_id_2, :].sketch_url[1]; #url for Q2
+		text_answer_q_id_2 = student_answers[student_answers.q_id .== q_id_2, :].text_answer[1]; #ans for Q2
 	end
+	 #filtered sketch data frame containing the entries of the selected student
+	sketch_data = filter(row -> row.uid == selected_student, s_df)
+	#building the dataset containing the timestamp vs y-vectors at each timestamp for each student
+	sy_data_1 , sy_data_2 = build_sy_dataset(
+		a_df[a_df.uid .== selected_student, :], 
+		s_df[s_df.uid .== selected_student, :])  
+	#filtered y vectors for selected student
+	animation_data_1=sy_data_1[sy_data_1.uid  .== selected_student, :] 
```

Comment:
Now that you are filtering `a_df` and `s_df` for the selected student on lines 231 and 232, `animation_data_1` is equal to `sy_data_1` and same for `animation_data_2` and `sy_data_2`. So line 234 and 235 are not needed and you can just call the variables on line 230 `animation_data_1` and `animation_data_2`.